nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/PutHiveStreaming.java

```diff
@@ -377,7 +377,7 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
         final AtomicInteger successfulRecordCount = new AtomicInteger(0);
         List<HiveStreamingRecord> successfulRecords = new LinkedList<>();
         final FlowFile inputFlowFile = flowFile;
-        final AtomicBoolean incomingFlowFileTransferred = new AtomicBoolean(false);
+        final AtomicBoolean processingFailure = new AtomicBoolean(false);
```

Comment:
@joewitt so in updating MockProcessSession to support the session migration, I found that the MockProcessSession was not properly keeping track of the 'recursionSet' that StandardProcessSession was. I fixed that bug, as it had to be addressed in order to properly implement the session migration in the Mock framework. Once I fixed that bug in Mock ProcessSession, it then exposed this unrelated bug in PutHiveStreaming. However, the bug in PutHiveStreaming would now cause unit test failures, I addressed that in this PR as well.